ledger/eval.go

```diff
@@ -211,13 +324,18 @@ type BlockEvaluator struct {
 }
 
 type ledgerForEvaluator interface {
+	ledgerForCowBase
 	GenesisHash() crypto.Digest
-	BlockHdr(basics.Round) (bookkeeping.BlockHeader, error)
 	Totals(basics.Round) (AccountTotals, error)
-	checkDup(config.ConsensusParams, basics.Round, basics.Round, basics.Round, transactions.Txid, txlease) error
+	CompactCertVoters(basics.Round) (*VotersForRound, error)
+}
+
+// ledgerForCowBase represents subset of Ledger functionality needed for cow business
+type ledgerForCowBase interface {
```

Comment:
what's the reason for splitting the interface ? is it for the ease of writing unit tests ?

Reply:
It is for ledger creation in tealdbg and dryrun. See `MakeDebugBalances`